src/layout/actions/layout.rs

```diff
@@ -901,18 +924,41 @@ impl LayoutTree {
             let parent_ix = self.tree.parent_of(child_ix)
                 .expect("Node had no parent");
             let children = self.tree.children_of(parent_ix);
-            let index = children.iter().position(|&node_ix| node_ix == child_ix)
-                .map(|num| (num + 1).to_string());
+            let index = children.iter().position(|&node_ix| node_ix == child_ix);
+            let index_str = index.map(|num| (num + 1).to_string());
+
+            // I can't do it inside the match ahead because it borrows
+            // self.tree, and I need self.tree to get the children's titles
+            let titles = children.iter().map(|&node_ix| {
+                match self.tree[node_ix] {
+                    Container::Container { ref mut borders, .. } =>
+                        borders.as_ref().map( |b| b.title() )
+                            .unwrap_or("").into(),
+                    Container::View { ref mut borders, .. } =>
+                        borders.as_ref().map( |b| b.title() )
+                            .unwrap_or("").into(),
```

Comment:
Because `borders` is the same for both, you can actually combine these two match statements to make the much more succinct:

```rust
                    Container::Container { ref mut borders, .. } |
                    Container::View { ref mut borders, .. } =>
                        borders.as_ref().map( |b| b.title() )
                            .unwrap_or("").into(),
```